### PR TITLE
Python3 Compatability

### PR DESCRIPTION
**Pull request by jonathanlunt**
- All current tests pass in python3.6
- Added python3.6 to tox configuration

### CODE REVIEW

File: disdat/hyperframe.py [hunk 1]
```diff
@@ -526,7 +528,7 @@ def detect_local_fs_path(series):
     """
     output = []
     for s in series:
-        if not isinstance(s, str) and not isinstance(s, unicode):
+        if not isinstance(s, str) and not isinstance(s, str):
```
Comment (kyocum) [inline disdat/hyperframe.py]: Well, that's not right. 
```suggestion
        if not isinstance(s, str):
```

File: disdat/hyperframe.py [hunk 2]
```diff
@@ -1163,7 +1170,7 @@ def get_frame_ids(self, names=None):
             (:obj:list (str,str))
         """
         if names is None:
-            names = [n for n in self.frame_dict.keys()]
+            names = [n for n in list(self.frame_dict.keys())]
```
Comment (kyocum) [inline disdat/hyperframe.py]:
```suggestion
            names = list(self.frame_dict.keys())
```

File: disdat/hyperframe.py [hunk 3]
```diff
@@ -1822,8 +1829,8 @@ def from_ndarray(hfid, name, nda):
             if all(isinstance(x, str) for x in nda):
                 frame_type = FrameRecord.get_proto_type(str)
                 series_data = nda
-            elif all(isinstance(x, unicode) for x in nda):
-                frame_type = FrameRecord.get_proto_type(unicode)
+            elif all(isinstance(x, str) for x in nda):
```
Comment (kyocum) [inline disdat/hyperframe.py]:
```suggestion
            elif all(isinstance(x, six.binary_type) for x in nda):
            frame_type=FrameRecord.get_proto_type(six.binarytype)
```

File: disdat/pipe.py [hunk 4]
```diff
@@ -211,7 +211,7 @@ def requires(self):
         for user_arg_name, cls_and_params in rslt.items():
             pipe_class, params = cls_and_params[0], cls_and_params[1]
 
-            if isinstance(pipe_class, str) or isinstance(pipe_class, unicode):
+            if isinstance(pipe_class, str):
```
Comment (kyocum) [inline disdat/pipe.py]:
```suggestion
            if isinstance(pipe_class, six.string_types):
```

File: disdat/pipe.py [hunk 5]
```diff
@@ -327,7 +327,7 @@ def _get_subcls_params(cls, self):
             if not isinstance(param_obj, luigi.Parameter):
                 continue
             val = getattr(self,p)
-            if type(val) is str or type(val) is unicode:
+            if type(val) is str:
```
Comment (kyocum) [inline disdat/pipe.py]: six.string_types

File: disdat/hyperframe.py (hunk 1 above)
Comment (kyocum) [reply disdat/hyperframe.py]: Fixed

File: disdat/hyperframe.py (hunk 2 above)
Comment (kyocum) [reply disdat/hyperframe.py]: Fixed

File: disdat/pipe.py (hunk 5 above)
Comment (kyocum) [reply disdat/pipe.py]: Done

**Comment by jonathanlunt** [comment]
@kyocum Fixed the weird type casts

**Comment by kyocum** [comment]
Created a kyocum/python3 branch in origin.

**Pull request closed without merging**